### PR TITLE
Add new retailers + fix several bugs with region selection

### LINKED ISSUE #92: Add Mosgiel Supermarkets
Supermarkets for Mosgiel are currently missing, despite Mosgiel being a part of Dunedin.

Related survey response:
*If not, what shops would you like to see in the app?*
>I said yes, but I'm not sure if Mosgiel New World was there.

### PR DESCRIPTION
**Pull request by SheaSmith**
Fixes #92, and some other bugs/features that didn't have issues.

### CODE REVIEW

File: scraper/src/main/java/com/example/cosc345/scraper/scrapers/generic/ShopifyScraper.kt [hunk 1]
```diff
@@ -22,7 +24,8 @@ import com.example.cosc345.shared.models.*
 abstract class ShopifyScraper(
```
Comment (petrafisher-code) [inline scraper/src/main/java/com/example/cosc345/scraper/scrapers/generic/ShopifyScraper.kt:24]: In the comment above, the line under the Process header is very long. Perhaps format that onto multiple lines.

File: scraper/src/main/java/com/example/cosc345/scraper/scrapers/woocommerce/KingsFishMarketScraper.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.example.cosc345.scraper.scrapers.woocommerce
+
+import com.example.cosc345.scraper.scrapers.generic.WooCommerceScraper
+import com.example.cosc345.shared.models.Region
+import com.example.cosc345.shared.models.Retailer
+import com.example.cosc345.shared.models.Store
+
+/**
+ * The Kings Fish Market specific implementation of the [WooCommerceScraper], which essentially just passes information through to it.
+ */
```
Comment (petrafisher-code) [inline scraper/src/main/java/com/example/cosc345/scraper/scrapers/woocommerce/KingsFishMarketScraper.kt:10]: @constructor Create a new instance of this scraper.

File: scraper/src/main/java/com/example/cosc345/scraper/scrapers/woocommerce/TastiBitzScraper.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package com.example.cosc345.scraper.scrapers.woocommerce
+
+import com.example.cosc345.scraper.scrapers.generic.WooCommerceScraper
+import com.example.cosc345.shared.models.Region
+import com.example.cosc345.shared.models.Retailer
+import com.example.cosc345.shared.models.Store
+
+/**
+ * The Tasti Bitz specific implementation of the [WooCommerceScraper], which essentially just passes information through to it.
+ */
```
Comment (petrafisher-code) [inline scraper/src/main/java/com/example/cosc345/scraper/scrapers/woocommerce/TastiBitzScraper.kt:10]: @constructor Create a new instance of this scraper.

**Review by petrafisher-code — CHANGES_REQUESTED** [review]
See comments, please. Love your work!

File: scraper/src/main/java/com/example/cosc345/scraper/scrapers/woocommerce/KingsFishMarketScraper.kt (hunk 2 above)
Comment (SheaSmith) [reply scraper/src/main/java/com/example/cosc345/scraper/scrapers/woocommerce/KingsFishMarketScraper.kt:10]: Was something I did initially, but really it wasn't needed and just bloats up the comments

File: scraper/src/main/java/com/example/cosc345/scraper/scrapers/woocommerce/TastiBitzScraper.kt (hunk 3 above)
Comment (SheaSmith) [reply scraper/src/main/java/com/example/cosc345/scraper/scrapers/woocommerce/TastiBitzScraper.kt:10]: See above

File: scraper/src/main/java/com/example/cosc345/scraper/scrapers/woocommerce/KingsFishMarketScraper.kt (hunk 2 above)
Comment (petrafisher-code) [reply scraper/src/main/java/com/example/cosc345/scraper/scrapers/woocommerce/KingsFishMarketScraper.kt:10]: Fair

File: scraper/src/main/java/com/example/cosc345/scraper/scrapers/generic/ShopifyScraper.kt (hunk 1 above)
Comment (SheaSmith) [reply scraper/src/main/java/com/example/cosc345/scraper/scrapers/generic/ShopifyScraper.kt:24]: Android Studios reformat doesn't seem to have a problem with it, so I guess it technically complies with the quasi-code standards for this project.

Comment (SheaSmith) [reply scraper/src/main/java/com/example/cosc345/scraper/scrapers/generic/ShopifyScraper.kt:24]: Might also mess with Markdown formatting in the compiled docs

Comment (code5920) [reply scraper/src/main/java/com/example/cosc345/scraper/scrapers/generic/ShopifyScraper.kt:24]: (Ew.)

Comment (petrafisher-code) [reply scraper/src/main/java/com/example/cosc345/scraper/scrapers/generic/ShopifyScraper.kt:24]: Ew.

**Review by petrafisher-code — APPROVED** [review]
Submitting my (begrudging) approval.

**Pull request merged**